zerver/models/pinned_views.py

```diff
@@ -0,0 +1,35 @@
+from enum import IntEnum
+from typing import Any
+
+from django.db import models
+from django.db.models import CASCADE
+
+from zerver.models.users import UserProfile
+
+
+class LeftSidebarViewLocationEnum(IntEnum):
+    MENU = 1
+    EXPANDED = 2
+
+
+class PinnedView(models.Model):
+    """
+    Represents a user's configuration for pinned views in the left sidebar, allowing views to be pinned or hidden.
+    """
+
+    user_profile = models.ForeignKey(UserProfile, on_delete=CASCADE)
```

Comment:
Let's call this `user = `; we will likely eventually rename `UserProfile` to `User`.